### ISSUE TITLE
C0103: Class name "__getattr__" doesn't conform to PascalCase naming style (invalid-name)

### ISSUE THREAD

**Issue by stdedos** [pylint-dev/pylint#9048]
### Bug description

https://github.com/hoel-bagard/matplotlib-stubs/blob/f24bcb7808881bf5a1dd7271ce50b0ad0fb0051f/src/matplotlib-stubs/__init__.pyi#L1-L12

### Command used

```shell
pylint --py-version 3.8 src/matplotlib-stubs/
```

### Pylint output

```shell
************* Module 
src/matplotlib-stubs/__init__.pyi:1:0: C0103: Module name "" doesn't conform to snake_case naming style (invalid-name)
src/matplotlib-stubs/__init__.pyi:9:0: C0414: Import alias does not rename original package (useless-import-alias)
src/matplotlib-stubs/__init__.pyi:9:0: W0406: Module import itself (import-self)
src/matplotlib-stubs/__init__.pyi:9:0: E0611: No name 'colors' in module '' (no-name-in-module)
src/matplotlib-stubs/__init__.pyi:10:0: E0401: Unable to import '_typing' (import-error)
src/matplotlib-stubs/__init__.pyi:12:0: C0103: Class name "__getattr__" doesn't conform to PascalCase naming style (invalid-name)
```

### Expected behavior

```
************* Module 
src/matplotlib-stubs/__init__.pyi:9:0: C0414: Import alias does not rename original package (useless-import-alias)
src/matplotlib-stubs/__init__.pyi:9:0: W0406: Module import itself (import-self)
```

and I am not sure about that W0406

### Pylint version

```shell
pylint 3.0.0a8-dev0
astroid 3.0.0a9
Python 3.8.10 (default, May 26 2023, 14:05:08) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04

### Additional dependencies

```
ansi2html==1.8.0
astroid==3.0.0a9
asttokens==2.4.0
azure-core==1.24.0
azure-storage-file-share==12.4.1
backcall==0.2.0
build==0.10.0
certifi==2020.12.5
cffi==1.14.5
cfgv==3.3.1
chardet==4.0.0
charset-normalizer==3.2.0
click==8.1.3
colorhash==1.2.1
colour==0.1.5
contourpy==1.1.0
cryptography==3.4.6
cycler==0.11.0
dash==2.13.0
dash-bootstrap-components==1.3.1
dash-core-components==2.0.0
dash-cytoscape==0.3.0
dash-html-components==2.0.0
dash-renderer==1.8.3
dash-renderjson==0.0.1
dash-table==5.0.0
decorator==5.1.1
dill==0.3.7
distlib==0.3.6
exceptiongroup==1.1.3
executing==1.2.0
filelock==3.12.2
Flask==2.2.5
flask-healthz==0.0.3
fonttools==4.40.0
identify==2.5.24
idna==2.10
importlib-metadata==6.8.0
importlib-resources==6.0.1
iniconfig==2.0.0
ipython==8.12.2
isodate==0.6.0
isort==5.12.0
itsdangerous==2.1.2
jedi==0.19.0
Jinja2==3.1.2
jq==1.5.0
kiwisolver==1.4.4
lazy-object-proxy==1.9.0
MarkupSafe==2.1.3
matplotlib==3.7.1
matplotlib-inline==0.1.6
matplotlib-stubs @ file:///repos/matplotlib-stubs
mccabe==0.7.0
msrest==0.6.21
mypy==1.5.1
mypy-extensions==1.0.0
nest-asyncio==1.5.7
nodeenv==1.8.0
numpy==1.24.4
oauthlib==3.1.0
packaging==23.1
pandas==2.0.3
pandas-stubs==2.0.3.230814
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.5.0
pip-tools==6.13.0
platformdirs==3.8.0
plotly==5.16.1
pluggy==1.3.0
pre-commit==3.3.3
prompt-toolkit==3.0.39
ptyprocess==0.7.0
pudb==2022.1.3
pure-eval==0.2.2
pycparser==2.20
Pygments==2.16.1
pylint @ git+https://github.com/pylint-dev/pylint@aba10dc7d197b01e7a683c326818621addf076cb
pyparsing==3.1.0
pyproject_hooks==1.0.0
pyright==1.1.315
pytest==7.4.1
pytest-mypy-plugins==3.0.0
pytest-mypy-testing==0.1.1
pytest-pudb==0.7.0
python-dateutil==2.8.2
pytz==2023.3
PyYAML==6.0
regex==2023.8.8
requests==2.25.1
requests-oauthlib==1.3.0
retrying==1.3.4
ruff==0.0.275
seaborn==0.12.2
six==1.15.0
stack-data==0.6.2
tenacity==8.2.3
tomli==2.0.1
tomlkit==0.12.1
traitlets==5.10.0
types-pytz==2023.3.0.1
typing_extensions==4.7.1
tzdata==2023.3
urllib3==1.26.4
urwid==2.1.2
urwid-readline==0.13
virtualenv==20.23.1
wcwidth==0.2.6
Werkzeug==2.2.3
wrapt==1.15.0
zipp==3.16.2
```

**Comment by mbyrnepr2** [comment]
A class with this name is expected to emit that Pylint `invalid-name` message; so that looks ok to me (either in a .py or .pyi file). Do I understand the ask of this issue correctly?

example.py:
```python
class __getattr__:
    ...
```

pylint example.py
```python
C0103: Class name "__getattr__" doesn't conform to PascalCase naming style (invalid-name)
```

**Comment by stdedos** [comment]
Oops 😅 I forgot to mention the https://peps.python.org/pep-0562/

tl;dr: Yes, technically it fails - but it is part of the PEP
I'd say it should be explicitly ignored, no-user-conf-required

**Comment by mbyrnepr2** [comment]
@stdedos thanks for that and no problem :).
Do I understand correct - this is for functions and not classes? Pylint will be happy if the name is used for a function. E.g:
```python
from typing import Any

def __getattr__(name: str) -> Any: ...
```
